content/en/blog/2022/k8sattributes.md

```diff
@@ -0,0 +1,261 @@
+---
+title: Improved troubleshooting using k8s attributes
+linkTitle: Kubernetes attributes
+date: 2022-06-27
+author: Ruben Vargas
+spelling:
+  cSpell:ignore k8sattributes k8sattributesprocessor K8sattributes k8sprocessor
+  cSpell:ignore K8sprocessor KUBE
+---
+
+Attaching Kubernetes resource metadata to OpenTelemetry traces can be very
+useful because it lets you identify which resource (such as a pod) is failing or
+having performance problems. It is also generally useful for correlating across
+other signals.
+
+In this article, you'll learn how to configure the OpenTelemetry collector to
+use the `k8sattributesprocessor` processor in different scenarios. At the end,
+we present some alternatives.
+
+Details of the OpenTelemetry collector pipeline won't be covered in this post.
+For those details please, refer to
+[OpenTelemetry collector documentation](/docs/collector/).
+
+## How k8s attributes are attached
+
+At a high level, k8s attributes are attached to traces as
+[resources](/docs/concepts/glossary/#resource). This is for two reasons:
+
+1.  K8s attributes fit the definition of what a resource is -- an entity for
+    which telemetry is recorded
+2.  It centralizes this metadata, which is relevant for any generated span.
+
+Let's dive in and see how to do it!
+
+## Using k8sattributes processor
+
+The `k8sattributesprocessor` is an OpenTelemetry collector processor that
+attaches Kubernetes attributes to spans, logs and metrics.
+
+This processor automatically discovers pod attributes and attaches those
+attributes to a resource associated with the spans generated by that pod. If the
+pod belongs to a `Deployment` or a `ReplicaSet`, it will also discover it's
+attributes.
+
+Some attributes we can attach to the resource are:
+
+- Node name `k8s.node.name`
+- Pod name `k8s.pod.name`
+- Pod UID `k8s.pod.uid`
+- Namespace `k8s.namespace.name`
+- Deployment name, if the pod was created by a deployment
+
+Such attributes adhere to OpenTelemetry semantic conventions. For details, see
+the [Kubernetes resource semantic conventions][].
+
+The processor internally maintains a list of pods and an associated attribute,
+usually the IP address of the pod, and uses this attribute to know which pod
+generates a certain span.
+
+![k8sattributes processor data flow](/img/blog-k8sattributes/k8sprocessor.png)
```

Comment:
In the diagram, "Add attributes" names an action. IMHO, it doesn't make sense to have an entity in this diagram representing an action. Or am I misunderstanding something here?

Reply:
I can't find a better way to represent it :/

Reply:
Consider just dropping the "Add attributes" box. The figure is meant to represent "how the data flows".